main/java/mjc/Runner.java

```diff
@@ -0,0 +1,184 @@
+        Map<AbstractPurchase, Weekday> purchasesFirstDayMap;
+        Map<Weekday, PurchaseList> weekdayPurchaseEnumMap;
+        PurchaseList discountedPurchases;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILENAME))) {
+            purchasesLastDayMap = new HashMap<>();
+            purchasesFirstDayMap = new HashMap<>();
+            weekdayPurchaseEnumMap = new EnumMap<>(Weekday.class);
+            discountedPurchases = new PurchaseList();
+            String day, csvLine;
+
+            while ((csvLine = reader.readLine()) != null && (day = reader.readLine()) != null) {
+                AbstractPurchase purchase = PurchaseFactory.getPurchase(csvLine);
+                Weekday weekday = Weekday.valueOf(day);
+
+                putLastDay(purchasesLastDayMap, weekday, purchase);
+                putFirstDay(purchasesFirstDayMap, weekday, purchase);
+                putEnumMapPurchase(weekdayPurchaseEnumMap, weekday, purchase);
+                if (purchase instanceof PurchaseAllDiscounted) {
+                    discountedPurchases.add(purchase);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("No source file found.");
+        } catch (CsvLineException e) {
+            throw new RuntimeException();
+        }
+
+        System.out.println("Purchase => Weekday of LAST purchase");
+        printMap(purchasesLastDayMap);
+        System.out.println("Purchase => Weekday of FIRST purchase");
+        printMap(purchasesFirstDayMap);
+
+        Purchase purchase1 = new Purchase("bread", 155, 1);
+        System.out.println("Find FIRST weekday for product " + purchase1.getProduct() + ": " + findMapValue(purchasesFirstDayMap, purchase1));
+        System.out.println("Find LAST weekday for product " + purchase1.getProduct() + ": " + findMapValue(purchasesLastDayMap, purchase1));
+        Purchase purchase2 = new Purchase("bread", 170, 1);
+        System.out.println("Find FIRST weekday for product " + purchase2.getProduct() + ": " + findMapValue(purchasesFirstDayMap, purchase2));
+
+        removeMapEntries(purchasesLastDayMap, new EntryChecker<>() {
+            @Override
+            public boolean check(Map.Entry<AbstractPurchase, Weekday> entry) {
+                return entry.getKey().getProduct().getName().equals("meat");
+            }
+        });
+        removeMapEntries(purchasesFirstDayMap, new EntryChecker<>() {
+            @Override
+            public boolean check(Map.Entry<AbstractPurchase, Weekday> entry) {
+                return entry.getValue().equals(Weekday.FRIDAY);
+            }
+        });
+
+        System.out.println("\nRemoved entries where name is 'meat'");
+        printMap(purchasesLastDayMap);
+        System.out.println("Removed entries where weekday is 'FRIDAY'");
+        printMap(purchasesFirstDayMap);
+
+        System.out.println("Total cost of discounted purchases: " + discountedPurchases.getTotalCost() + "\n");

```

Comment:
breaking the task formulation. 
create static method `getTotalCost` of the `Runner`.